sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt

```diff
@@ -401,30 +398,38 @@ class OfflineTest {
     Assert.assertEquals(0, stylePackWithNoStyle.first.size)
   }
 
-  @Ignore("TODO uncomment after https://github.com/mapbox/mapbox-maps-android/issues/297")
+  // It's still discussable whether we should trigger MAP_IDLE event
+  // if there's no internet and Mapbox stack is set to True.
+  // Currently it works as designed - we do not trigger MAP_IDLE if any
+  // errors occurred during style load.
   @Test
   fun idleEventTileStoreAirplaneMode() {
     loadTileStoreWithStylePack()
     val latch = CountDownLatch(1)
     var idleEventCount = 0
+    var mapLoadingErrorCount = 0
     val observer = object : Observer() {
       override fun notify(event: Event) {
         Logger.e(TAG, "type ${event.type}, data ${event.data.toJson()}")
         if (event.type == MapEvents.MAP_IDLE) {
           idleEventCount++
         }
+        if (event.type == MapEvents.MAP_LOADING_ERROR) {
+          mapLoadingErrorCount++
+        }
       }
     }
     switchAirplaneMode()
     verifyAirplaneModeEnabled()
     prepareMapView(
       observer,
-      listOf(MapEvents.MAP_IDLE)
+      listOf(MapEvents.MAP_IDLE, MapEvents.MAP_LOADING_ERROR)
     )
     try {
       // we do not throw timeout exception and verify we have exactly 1 IDLE event after timeout
       latch.await(10, TimeUnit.SECONDS)
-      Assert.assertEquals(1, idleEventCount)
+      MatcherAssert.assertThat(mapLoadingErrorCount, Matchers.greaterThan(0))
```

Comment:
Is this code reachable if timeout? Guess `latch.await` through timeout exception and go into `finally` directly.

Reply:
Yes, it is reachable, we simply resume execution after latch has timed out.

Reply:
Timeout exception happens if we throw it:
```
if (!latch.await(10, TimeUnit.SECONDS)) {
  throw TimeoutException()
}
```